projects/plugins/jetpack/_inc/client/components/connect-button/index.jsx

```diff
@@ -224,10 +237,20 @@ export class ConnectButton extends React.Component {
 				) }
 				{ this.renderContent() }
 				{ this.props.children }
-				<JetpackDisconnectModal
-					show={ this.state.showModal }
-					showSurvey={ false }
-					toggleModal={ this.toggleVisibility }
+				<DisconnectDialog
+					apiNonce={ this.props.apiNonce }
+					apiRoot={ this.props.apiRoot }
+					connectedPlugins={ this.props.connectedPlugins }
+					connectedUser={ {
+						ID: this.props.userWpComId,
+						login: this.props.userWpComLogin,
+					} }
+					connectedSiteId={ this.props.connectedSiteId }
```

Comment:
I am not super clear on why we need this; is this something that was added in another PR? Since we have the site URL, wouldn't that be an option here, something we can build upon from the  data we have on WordPress.com?

Reply:
This is related to the changes introduced in #21048 . The `<DisconnectDialog />` component handles collecting the optional user survey that is part of the updated disconnection flow. The `connectedUser.ID` and `connectedSiteId` values are [used when forming the survey response](https://github.com/Automattic/jetpack/pull/21048/files#diff-a71b0372aa871fa3f40fbc9fd0ed6ffc8da3f01e3242d3db623a788ff7a7eaefR247-R257). 
Open to feedback or alternative approaches if there is a better way to handle the survey.

Reply:
Ok, that makes sense, thank you!